documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -111,35 +118,156 @@ for fuzzy matches.
 **Finally**, we apply the ranking and sorting parameters, and "highlight" the
 fields that were matched in the results.
 
-The new search controller should allow for using different filters for the first
-step and to not use the full-text search. We should also create a new serializer
-for collection search requests. It should include the common parameters for
-`list` requests, such as `page` and `page_size`, and the parameters for the
-exact matches: `tag`, `creator` and `source`.
+The search controller needs to be updated to extract the first 2 steps into
+`build_search_query` method. A new `build_collection_query` should be added and
+used when the `collection` parameter is present. This method should create a
+filter query for the relevant field and value.
+
+The pagination and dead link cleanup should be the same for additional search
+views as for the default search views.
+
+#### Search request serializer updates
+
+_The initial version of this plan proposed to add new API endpoints and use path
+parameters for the values (e.g., `/image/tag/cat`) to make the URLs more
+readable, easier to share, will be easier to cache or perform cache invalidation
+required by [#1969](https://github.com/WordPress/openverse/issues/1969).
+However, since tags and creator names contain characters that are special for
+path segments (`/`, `?` and `&`) that cannot be properly encoded, this approach
+turned out to be not feasible._
+
+```{note}
+The new query parameters (`collection` and `tag`) will have `unstable__` prefix until this project is
+launched. In the text below, for brevity, the prefix is omitted.
+```
+
+The collections will use the `search` endpoint with `collection` query parameter
+set to `tag`, `creator` or `source`, and the relevant query parameters. The
+`collection` parameter validator will check that the request contains the
+necessary additional parameters: `source` for `collection=source`, `creator` and
+`source` for `collection=creator`, and `tag` for `collection=tag`. The value of
+the `source` parameter will also be checked to be an existing source using the
+provider store.
+
+For the tag route, the new singular `tag` parameter should be used, rather than
+the existing plural `tags` should be used for legibility since we are presenting
+a single tag.
+
+The existing `source` and `creator` parameters will be reused, but will be
+parsed differently when `collection` parameter is present: they will only allow
+a single value instead of being split by `,` as it is for the default search.
```

Comment:
Does "allow" just mean that these parameters won't perform the splitting, and instead will read the strings as-is? So `source=flickr,europeana` would search for `flickr,europeana`? As opposed to say, throwing an error or selecting the first item in the list (flickr)?

I just want to make sure we're specific with what "allow" means.

Reply:
> Does "allow" just mean that these parameters won't perform the splitting, and instead will read the strings as-is?

Yes, it will read the strings as-is, without splitting by comma.

I was thinking that the API would throw an error if the source is not in the list of sources for the media type. So, for instance, `source=flickr,europeana` would throw an error if used with `collection=source`. The error would say something like "Invalid source parameter for source collection: `flickr,europeana`".

Do you think it's the most clear way of handling such requests? Or should we rather use the first source?

Reply:
That sounds great! I think it could be documented just like that in the document.